mobile/src/main/java/org/openhab/habdroid/ui/ItemPickerAdapter.java

```diff
@@ -0,0 +1,174 @@
+package org.openhab.habdroid.ui;
+
+import android.content.Context;
+import android.net.Uri;
+import android.preference.PreferenceManager;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.core.graphics.drawable.DrawableCompat;
+import androidx.recyclerview.widget.RecyclerView;
+
+import org.openhab.habdroid.R;
+import org.openhab.habdroid.core.connection.Connection;
+import org.openhab.habdroid.core.connection.ConnectionFactory;
+import org.openhab.habdroid.core.connection.exception.ConnectionException;
+import org.openhab.habdroid.model.Item;
+import org.openhab.habdroid.ui.widget.WidgetImageView;
+import org.openhab.habdroid.util.Constants;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+
+public class ItemPickerAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder>
+        implements View.OnClickListener {
+    public interface ItemClickListener {
+        void onItemClicked(Item item);
+    }
+
+    private ArrayList<Item> mItems = new ArrayList<>();
+    private ArrayList<Item> mAllItems = new ArrayList<>();
+    private ItemClickListener mItemClickListener;
+    private final LayoutInflater mInflater;
+    private int mHighlightedPosition = -1;
+    private static String mIconFormat;
+
+    public ItemPickerAdapter(Context context, ItemClickListener itemClickListener) {
+        super();
+        mIconFormat = PreferenceManager.getDefaultSharedPreferences(context)
+                .getString(Constants.PREFERENCE_ICON_FORMAT, "");
+        mInflater = LayoutInflater.from(context);
+        mItemClickListener = itemClickListener;
+    }
+
+    public void addLoadedItems(List<Item> items) {
+        mItems.clear();
+        mItems.addAll(items);
+        Collections.sort(mItems, new ItemNameComparator());
+        mAllItems = new ArrayList<>(mItems);
```

Comment:
Why not `clear()` + `addAll()`?

Reply:
Because I already call clear(), addAll() and sort() on mItems. After that IMO copying is smarter than calling the three functions again.

Reply:
I didn't say you should re-sort, but do `addAll(mItems)` ;-) I think clear + addAll is better because it avoids the memory reallocation.

Reply:
I thought you mean `mAllItems.addAll(items)`.